payments/tests/test_payments_models.py

```diff
@@ -387,6 +389,62 @@ def test_order_berth_lease_right_price_for_full_season(harbor):
             assert product_price == order_price
 
 
+def test_order_winter_storage_lease_right_price_for_full_season(winter_storage_area):
+    services = {
+        "summer_storage_for_docking_equipment": True,
+        "summer_storage_for_trailers": random_bool(),
+    }
+    for service, create in services.items():
+        if create:
+            # Using PriceUnits.AMOUNT to simplify testing
+            AdditionalProductFactory(
+                service=ProductServiceType(service), price_unit=PriceUnits.AMOUNT
+            )
+
+    section = WinterStorageSectionFactory(**services, area=winter_storage_area)
+    lease = WinterStorageLeaseFactory(
+        place=WinterStoragePlaceFactory(winter_storage_section=section)
+    )
+    product = WinterStorageProductFactory(winter_storage_area=winter_storage_area)
+    order = Order.objects.create(product=product, customer=lease.customer, lease=lease)
+
+    for service, create in services.items():
+        if create:
+            additional_product = AdditionalProduct.objects.filter(
+                service=ProductServiceType(service),
+                price_unit=PriceUnits.AMOUNT,
+                period=PeriodType.SEASON,
+            ).first()
+            OrderLine.objects.create(order=order, product=additional_product)
+
+    assert order.lease.start_date == calculate_winter_storage_lease_start_date()
+    assert order.lease.end_date == calculate_winter_storage_lease_end_date()
+    sqm = order.lease.place.place_type.width * order.lease.place.place_type.length
+
+    expected_price = calculate_product_partial_season_price(
+        product.price_value,
+        calculate_winter_storage_lease_start_date(),
+        calculate_winter_storage_lease_end_date(),
+        summer_season=False,
+    )
+    expected_price = rounded(expected_price * sqm, decimals=2)
+    assert order.price == expected_price
```

Comment:
and doing this you just test that `calculate_product_partial_season_price` is used to calculate the price. a slightly more explicit approach to testing, I think, would be to: (1) freeze time and (2) compare `order.price` against some hardcoded `Decimal` value